public_html/core/engine/dispatcher.php

```diff
@@ -286,86 +271,95 @@ public function dispatch($parent_controller = '')
+                    );
+                    //Process each child and create dispatch to call recursive
+                    $dispatch = new ADispatcher($child['controller'], ["instance_id" => $child['instance_id']]);
+                    $dispatch->dispatch($controller);
+                    // Append output of child controller to current controller
+                    if (isset($child['position'])
+                        && $child['position']) { // made for recognizing few custom_blocks in the same placeholder
+                        $controller->view->assign(
+                            $child['block_txt_id'].'_'.$child['instance_id'],
+                            $this->response->getOutput()
+                        );
+                    } else {
+                        $controller->view->assign($child['block_txt_id'], $this->response->getOutput());
+                    }
+                    //clean up and remove output
+                    $this->response->setOutput('');
+                    ADebug::checkpoint($child['controller'].' ( child of '.$this->controller.' ) dispatch END');
+                }
+                //Request controller to generate output
+                $controller->finalize();
 
-            //add pre and post controllers output
-            $this->response->setOutput($output_pre.$this->response->getOutput().$output_post);
+                //check for controller.pre
+                $output_post = $this->dispatchPrePost($this->controller.POSTFIX_POST);
+                //add pre and post controllers output
+                $this->response->setOutput($output_pre.$this->response->getOutput().$output_post);
 
-            //clean up and destroy the object
-            unset($controller);
-            unset($dispatch);
-        } else {
-            $err = new AError(
-                'Error: controller method not exist '.$this->class.'::'.$this->method.'!',
-                AC_ERR_CLASS_METHOD_NOT_EXIST
-            );
-            $err->toLog()->toDebug();
-            if (in_array($this->controller_type, ['responses', 'api', 'task'])) {
-                $dd = new ADispatcher('responses/error/ajaxerror/not_found');
-                $dd->dispatch();
+                //clean up and destroy the object
+                unset($controller, $dispatch);
+            } else {
+                $err = new AError(
+                    'Error: controller method not exist '.$this->class.'::'.$this->method.'!',
+                    AC_ERR_CLASS_METHOD_NOT_EXIST
+                );
+                $err->toLog()->toDebug();
+                if (in_array($this->controller_type, ['responses', 'api', 'task'])) {
+                    $dd = new ADispatcher('responses/error/ajaxerror/not_found');
+                    $dd->dispatch();
+                }
+            }
+        } //catching output of around hook (it can be only one)
+        catch (AException $e) {
+            if ($e->getCode() != AC_HOOK_AROUND) {
```

Comment:
If this functionality takes entire control of the execution, it should be called takeover or bypass hook. We need to rename it in all places.